### PR TITLE
Use upstream library to create ProwJobs from config

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/assign @wking @smarterclayton 

We need to make this repo build with `go mod` at some point...

```
$ make build
go build -ldflags "-X github.com/openshift/release-controller/vendor/k8s.io/client-go/pkg/version.gitCommit=$(git rev-parse HEAD) -X github.com/openshift/release-controller/vendor/k8s.io/client-go/pkg/version.gitVersion=v1.0.0+$(git rev-parse --short=7 HEAD)" ./cmd/release-controller
...
build github.com/openshift/release-controller/cmd/release-controller: cannot load gopkg.in/russross/blackfriday.v2: cannot find module providing package gopkg.in/russross/blackfriday.v2
make: *** [Makefile:2: build] Error 1

```

### CODE REVIEW

File: cmd/release-controller/audit_backend_gcs.go [hunk 1]
```diff
@@ -126,7 +125,7 @@ func (b *GCSAuditStore) PutSignature(ctx context.Context, dgst string, signature
 		objectPath = path.Join("signatures", "openshift", "release", fmt.Sprintf("%s=%s", parts[0], parts[1]), fmt.Sprintf("signature-%d", index))
 	}
 	glog.V(4).Infof("Writing signature to gs://%s/%s", b.bucketName, objectPath)
-	obj := b.bucket.Object(objectPath).WithConditions(storage.IfGenerationMatch(0))
+	obj := b.bucket.Object(objectPath).If(storage.Conditions{GenerationNotMatch:1})
```
Comment (stevekuznetsov) [inline cmd/release-controller/audit_backend_gcs.go]: @smarterclayton please review this

File: .gitignore [hunk 2]
```diff
@@ -1 +1,2 @@
 /release-controller
+.idea/*
```
Comment (wking) [inline .gitignore]: Seems unrelated to Go modules?  I expect this falls under "Patterns which a user wants Git to ignore in all situations" in [`gitignore(1)`][1], which suggests putting these in a file that defaults to `$XDG_CONFIG_HOME/git/ignore`.

[1]: https://git-scm.com/docs/gitignore#_description

**Comment by smarterclayton** [comment]
/hold

7809 files, huh?

File: go.mod [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+	golang.org/x/time v0.0.0-20181108054448-85acf8d2951c
+	google.golang.org/api v0.10.0
+	gopkg.in/inf.v0 v0.9.0 // indirect
+	k8s.io/api v0.0.0-20190222213804-5cb15d344471
+	k8s.io/apimachinery v0.0.0-20190221213512-86fb29eff628
+	k8s.io/client-go v0.0.0-20190228174230-b40b2a5939e4
+	k8s.io/klog v0.0.0-20181108234604-8139d8cb77af
+	k8s.io/kube-openapi v0.0.0-20181109181836-c59034cc13d5 // indirect
+	sigs.k8s.io/yaml v1.1.0 // indirect
+)
```
Comment (bradmwilliams) [inline go.mod]: @stevekuznetsov If you add the following block, it will resolve the **log_dir** issue that is occurring with your changes:
```suggestion
)

replace (
          github.com/golang/glog => github.com/openshift/golang-glog 3c92600d7533018d216b534fe894ad60a1e6d5bf
)
```
Run a `tidy` and a `vendor` to pick up all the latest/greatest changes, and the release-controller should be operational again.

**Comment by stevekuznetsov** [comment]
I want a review on this: https://github.com/openshift/release-controller/pull/111#discussion_r323502498

File: cmd/release-controller/audit_backend_gcs.go [hunk 4]
```diff
@@ -4,6 +4,7 @@ import (
 	"context"
 	"crypto/sha256"
 	"fmt"
+	"google.golang.org/api/iterator"
```
Comment (smarterclayton) [inline cmd/release-controller/audit_backend_gcs.go]: wrong import block

File: cmd/release-controller/audit_backend_gcs.go (hunk 1 above)
Comment (smarterclayton) [reply cmd/release-controller/audit_backend_gcs.go]: Generation can be more than one if someone has edited the signature, is there a `storage.Conditions` for `GTE`?

**Comment by bradmwilliams** [comment]
@stevekuznetsov This PR can closed now, since #124 has been merged and appears to be running as expected.
